demos/text_to_speech_demo/python/text_to_speech_demo.py

```diff
@@ -49,39 +53,55 @@ def build_argparser():
     args.add_argument("-m_forward", "--model_forward",
                       help="Required. Path to ForwardTacotron`s mel-spectrogram regression part (*.xml format).",
                       required=True, type=str)
-    args.add_argument("-m_upsample", "--model_upsample",
-                      help="Required. Path to WaveRNN`s part for mel-spectrogram upsampling "
-                           "by time axis (*.xml format).",
-                      required=True, type=str)
-    args.add_argument("-m_rnn", "--model_rnn",
-                      help="Required. Path to WaveRNN`s part for waveform autoregression (*.xml format).",
-                      required=True, type=str)
-
     args.add_argument("-i", "--input", help="Text file with text.", required=True,
                       type=str)
     args.add_argument("-o", "--out", help="Required. Path to an output .wav file", default='out.wav',
                       type=str)
 
-    args.add_argument("--upsampler_width", default=-1,
-                      help="Width for reshaping of the model_upsample. "
-                           "If -1 then no reshape. Do not use with FP16 model.",
-                      required=False,
-                      type=int)
-
     args.add_argument("-d", "--device",
                       help="Optional. Specify the target device to infer on; CPU, GPU, FPGA, HDDL, MYRIAD or HETERO is "
                            "acceptable. The sample will look for a suitable plugin for device specified. "
                            "Default value is CPU",
                       default="CPU", type=str)
+
+    vocoder_parsers = parser.add_subparsers(dest='vocoder')
```

Comment:
Subcommands don't seem like the best choice for choosing the model architecture. For one, purely semantically the vocoder type isn't a subcommand. But also, consider what'll happen if in the future you need to add a similar selection for the mel-spectrogram generator. You can't specify two subcommands.

IMO, you should just add all possible model options to the main parser and then make sure that incompatible options aren't specified. `add_mutually_exclusive_group` can help with this.

Reply:
Done. As I understood from documentation add_mutually_exclusive_group is not good choice for groups like [m_melgan | m_rnn & m_upsample]. It is possible only split by  [m_melgan | m_rnn | m_upsample]

Reply:
What I meant was that you could make, for example, `m_melgan` and `m_rnn` mutually exclusive, and then check the presence of the other options based on which one was specified. The current approach is fine too, though.